mfs/system.go

```diff
@@ -123,6 +123,36 @@ func (kr *Root) Flush() error {
 	return nil
 }
 
+// FlushMemFree flushes the root directory and then uncaches all of its links.
+// This has the effect of clearing out potentially stale references and allows
+// them to be garbage collected.
+// CAUTION: Take care not to ever call this while holding a reference to any
+// child directories. Those directories will be bad references and using them
+// may have unintended racy side effects.
+// A better implemented mfs system (one that does smarter internal caching and
+// refcounting) shouldnt need this method.
+func (kr *Root) FlushMemFree(ctx context.Context) error {
+	dir, ok := kr.GetValue().(*Directory)
+	if !ok {
+		return fmt.Errorf("invalid mfs structure, root should be a directory")
+	}
+
+	if err := dir.Flush(); err != nil {
```

Comment:
What happens if we concurrently modify this directory? Can we end up uncaching non-flushed entries? Is that an issue?

Should we have a single `FlushAndUncache()` method?

Reply:
Thats basically the purpose of this method. In my comment i make it explicit that you shouldnt use this method concurrently with any other operations since there are deeper issues that could potentially happen.

Reply:
Fair enough. I'm working on improving this situation, FYI.